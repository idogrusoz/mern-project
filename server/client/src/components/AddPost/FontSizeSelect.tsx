import { MenuItem, Select } from "@material-ui/core";
import React, { Dispatch, FunctionComponent, SetStateAction } from "react";

type FontSizeSelectProps = {
    fontSize: number;
    setFontSize: Dispatch<SetStateAction<number>>;
};

const FontSizeSelect: FunctionComponent<FontSizeSelectProps> = ({ fontSize, setFontSize }) => {
    const fontSizes: number[] = [16, 20, 22, 24, 28, 32, 36, 40, 48, 56, 64, 72, 144];
    return (
        <Select fullWidth defaultValue={fontSize} onChange={(e) => setFontSize(Number(e.target.value))}>
            {fontSizes.map((size: number, i: number) => {
                return (
                    <MenuItem value={size} key={i}>
                        {size}
                    </MenuItem>
                );
            })}
        </Select>
    );
};

export default FontSizeSelect;
