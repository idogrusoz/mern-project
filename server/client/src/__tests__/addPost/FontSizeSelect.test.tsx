import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import FontSizeSelect from "../../components/addPost/FontSizeSelect";

describe("FontSizeSelect test", () => {
    const props = {
        fontSize: 40,
        setFontSize: jest.fn(() => {}),
    };

    const wrapper = shallow(<FontSizeSelect {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
