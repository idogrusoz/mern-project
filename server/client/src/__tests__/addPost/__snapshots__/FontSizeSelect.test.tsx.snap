// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontSizeSelect test renders without crash 1`] = `
<WithStyles(ForwardRef(Select))
  defaultValue={40}
  fullWidth={true}
  onChange={[Function]}
>
  <WithStyles(ForwardRef(MenuItem))
    key="0"
    value={16}
  >
    16
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="1"
    value={20}
  >
    20
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="2"
    value={22}
  >
    22
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="3"
    value={24}
  >
    24
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="4"
    value={28}
  >
    28
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="5"
    value={32}
  >
    32
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="6"
    value={36}
  >
    36
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="7"
    value={40}
  >
    40
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="8"
    value={48}
  >
    48
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="9"
    value={56}
  >
    56
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="10"
    value={64}
  >
    64
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="11"
    value={72}
  >
    72
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="12"
    value={144}
  >
    144
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Select))>
`;
